### PR TITLE
Added documentation for several files.

### PR DESCRIPTION
**Pull request by pumken**
Added documentation comments for `spatial.rs`, `lib.rs`, `gui.rs`, and `drawing.rs`. Also applied linting to the aforementioned files, plus `pong.rs`.

I didn't add doc comments for the other files mostly because I don't understand how they work; maybe I'll get around to doing that ¯\_(ツ)_/¯

### CODE REVIEW

**Review by lemunozm — DISMISSED** [review]

File: examples/pong.rs [hunk 1]
```diff
@@ -11,7 +11,8 @@ const PAD_HEIGHT: i32 = 3;
 
 struct PlayerState {
     pub position: Vec2,
-    pub direction: i32, // -1, 0, 1
+    pub direction: i32,
+    // -1, 0, 1
```
Comment (lemunozm) [inline examples/pong.rs]: This comment should apply to the direction. It was clippy automatically, I think.
```suggestion
    // -1, 0, 1
    pub direction: i32,
```

File: src/drawing.rs [hunk 2]
```diff
@@ -16,20 +23,58 @@ pub struct RectCharset {
...
+    ///
+    /// # Rectangle
+    ///
+    /// Using this charset will provide a rectangle that looks like this:
+    ///
+    /// ```
+    /// ╔══════╗
+    /// ║ruscii║
+    /// ╚══════╝
+    /// ```
```
Comment (lemunozm) [inline src/drawing.rs:66]: awesome! ❤️

**Review by lemunozm — COMMENTED** [review]
Great work adding all these inline docs. 🙌🏻 

Thank you so much!

File: examples/pong.rs (hunk 1 above)
Comment (pumken) [reply examples/pong.rs]: Probably!

**Review by lemunozm — APPROVED** [review]

**Comment by lemunozm** [comment]
I'll merge it once the conflicts are solved.

Thanks for all your contributions! 🙌🏻

**Review by lemunozm — APPROVED** [review]

**Pull request merged**